plutus-errors/src/Errors.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Errors (errors) where
+
+import Language.Haskell.TH
+import Language.Plutus.Common
+
+import qualified Language.PlutusIR.Error as PIR
+import qualified Language.PlutusIR.Parser as PIR
+import qualified Language.PlutusCore.Error as PLC
+import qualified Language.PlutusCore.DeBruijn as PLC
+import qualified Language.PlutusCore.Evaluation.Machine.Exception as PLC
+import qualified Language.PlutusCore.Evaluation.Machine.Cek as PLC
+import qualified Language.UntypedPlutusCore.Evaluation.Machine.Cek as PLCU
+import qualified  Language.PlutusTx.Code as PTX
+import qualified  Language.PlutusTx.Lift.Class as PTX
+import qualified Language.PlutusTx.Utils as PTX
+import qualified  Language.PlutusTx.Compiler.Error as PTX
+import qualified Language.PlutusTx.Compiler.Expr as PTX
+import qualified Language.PlutusTx.Compiler.Type as PTX
+import qualified Language.PlutusTx.Compiler.Kind as PTX
+import qualified Language.PlutusTx.Compiler.Builtins as PTX
+
+{- | A collection of error instances and their codes that are deprecated.
+
+When an error is deprecated (does not trigger anymore) and (some of) its dataconstructors has been removed,
+and in case the error is "exposed" to the public, then it is required that its "deprecated" constructors
+be "moved" and listed/errorCoded under the umbrella datatype `plutus-errors:Errors.DeprecatedErrors`.
+
+See NOTE [Error Codes of plutus errors]
+-}
+{-# DEPRECATED DeprecatedErrors "These errors and their error codes *should* not be thrown by any plutus code anymore" #-}
```

Comment:
This datatype is not actually deprecated, right? It represents deprecated things, but that's different.

Reply:
Well yes, but is for styling haddock and giving a note to the user.